spacewalk/setup/lib/Spacewalk/Setup.pm

```diff
@@ -1119,10 +1149,19 @@ sub postgresql_clear_db {
 
 sub postgresql_drop_reportdb {
         my $answers = shift;
-
-        system_debug('/usr/bin/uyuni-setup-reportdb', 'remove',
-            '--db='.$answers->{'report-db-name'}, '--user='.$answers->{'report-db-user'});
-        return 1;
+        if ($answers->{'external-db'}) {
+          system_debug('/usr/bin/uyuni-setup-reportdb', 'remove',
+              '--db='.$answers->{'report-db-name'}, '--user='.$answers->{'report-db-user'},
+                    "--external-host", $answers->{'report-db-host'},
+                    "--external-user", $answers->{'external-db-user'},
+                    "--external-password", $answers->{'external-db-password'});
+          return 1;         
+        }
+        else {  
+          system_debug('/usr/bin/uyuni-setup-reportdb', 'remove',
+              '--db='.$answers->{'report-db-name'}, '--user='.$answers->{'report-db-user'});
+          return 1;
+        }
```

Comment:
Example to combine this:
```
@cmd = ('/usr/bin/uyuni-setup-reportdb', 'remove', '--db='.$answers->{'report-db-name'},
   '--user='.$answers->{'report-db-user'});
if ($answers->{'external-db'}) {
    push @cmd, "--external-host", $answers->{'report-db-host'},
                    "--external-user", $answers->{'external-db-user'},
                    "--external-password", $answers->{'external-db-password'};
}
system_debug(@cmd);
return 1;
```
(not tested)

Reply:
Thank you for the suggestion. I will clean this up.

Reply:
fixed